docs: Add section in logs arch doc about Loki HA (#3572)
scality/metalk8s: docs/developer/architecture/logs.rst

  define loki storage sizing depending on retention, ingestion rate,
  average size of logs, number of streams, ...

Loki HA
_______

It is possible to provide a member list to Loki through its configuration,

> so that it is aware of the other Loki instances.

Reviewer: ```suggestion
so that it can discover the other Loki instances.
```

This way, it allows any instance to replicate, on some or all the other
instances (depending on the replication factor), every logs it receives.

> This configuration will be dynamically generated during installation depending

Reviewer: Can't we try to make it dynamic ? If I recall correctly it was possible to configure loki member list through a service discovery mechanism (I remember seeing a mention to etcd, so maybe kubernetes based service discovery is achievable, worth checking IMO).

Author: Yep, looks like it is possible to pass a headless service.
I'll give it a try and change that accordingly.

stuck, waiting to be able to send its data. This way, as we use the service
to talk to one instance which then replicates the data to other instances, if
a Loki instance fails, fluent-bit will not know about it and will keep running.

Since we use filesystem as Loki storage backend for the logs, we cannot
ensure we will have the data if at some point an instance is down or the
instance loses its volume.
It means, if we query for this period of time and we hit this very instance
we will have no data.
Note that, if the instance is only down for few minutes (depending on the
quantity of logs of the platform), it will be able to catch up what is still
in other ingesters memory and not written yet to the storage backend.

To solve this problem, we either need to be able to share this storage
between all the instances (e.g. NFS), but it is probably a bad idea, either we
need an extra component in front of the queriers, which takes care of querying
each instance, deduplicating the entries and returning them.
We have already considered the Loki Query Frontend, but its goal is only
to split queries, dispatch the load on the multiple instances and do some

> caching.

Reviewer 2: Which could come in handy if we want to implement sharding
